### PR TITLE
Group dropdown

### PR DESCRIPTION
**Pull request by andrew-cline**
# Description

Added logic for enabling and disabling groups in the main graphing dropdown selection for groups depending on compatibility with the selected unit and meters. 

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations
If a unit is selected before a group is selected then all the groups disappear from the dropdown.

### CODE REVIEW

File: src/client/app/containers/ChartDataSelectContainer.ts [hunk 1]
```diff
@@ -260,5 +259,81 @@ function getUnitCompatibility(compatibleUnits: Set<number>, incompatibleUnits: S
...
+
+/**
+ * Determines the compatibility of group in the redux state for display in dropdown
+ * @param {State} state - current redux state
+ * @return {SelectOption[]} an array of SelectOption
+ */
+export function getGroupCompatibilityForDropdown(state: State) {
+	const visibleGroup: number[] = [];
+	// Get all the groups that this user can see.
+	if (state.currentUser.profile?.role == 'admin') {
```
Comment (huss) [inline src/client/app/containers/ChartDataSelectContainer.ts]: Should this be ===?

**Review by huss — CHANGES_REQUESTED** [review]
There is one simple comment to address. I have also pushed a number of changes that the commit message describes. You probably want to review them to make sure they are okay.

Overall, this works well. There are a few features left to do: allow to unselect unit back to -99 which clears all meters/groups and update unit if unselect all meters/groups. I'm debating if I'm going to put in more fixes to this PR or not for these items. Right now I need to go on to other things and will come back to this.

File: src/client/app/containers/ChartDataSelectContainer.ts (hunk 1 above)
Comment (andrew-cline) [reply src/client/app/containers/ChartDataSelectContainer.ts]: Added with most recent commit.

**Review by huss — APPROVED** [review]
The change is great and everything works fine. From what this PR wanted to do it is ready to merge.

I am going to wait to merge to see how the other dropdown menus work and make sure no changes are needed to make them consistent. It is now likely that the fixes for the interaction between all the new menus will be a new PR and not changes to this (and the other) PRs.

**Comment by huss** [comment]
React hooks have now been integrated into OED. The work in this PR has been integrated into that work and moved forward. Thus, the current plan is to close this PR and move forward with a new one.

**Pull request merged**